src/include/gurt/telemetry_common.h

```diff
@@ -10,13 +10,122 @@
+#define D_TM_MILLISECOND		"ms"
+#define D_TM_MICROSECOND		"us"
+#define D_TM_NANOSECOND			"ns"
+
+/** Units of measure for storage */
+#define D_TM_BIT			"b"
+#define D_TM_BYTE			"B"
+
+#define D_TM_KILOBIT			"kbit"
+#define D_TM_KIBIBIT			"Kibit"
+#define D_TM_KILOBYTE			"kB"
+#define D_TM_KIBIBYTE			"KiB"
+
+#define D_TM_MEGABIT			"Mbit"
+#define D_TM_MEBIBIT			"Mibit"
+#define D_TM_MEGABYTE			"MB"
+#define D_TM_MEBIBYTE			"MiB"
+
+#define D_TM_GIGABIT			"Gbit"
+#define D_TM_GIGIBIT			"Gibit"
+#define D_TM_GIGABYTE			"GB"
+#define D_TM_GIGIBYTE			"GiB"
+
+#define D_TM_TERABIT			"Tbit"
+#define D_TM_TEBIBIT			"Tibit"
+#define D_TM_TERABYTE			"TB"
+#define D_TM_TEBIBYTE			"TiB"
+
+#define D_TM_PETABIT			"Pbit"
+#define D_TM_PEBIBIT			"Pibit"
+#define D_TM_PETABYTE			"PB"
+#define D_TM_PEBIBYTE			"PiB"
+
+#define D_TM_EXABIT			"Ebit"
+#define D_TM_EXBIBIT			"Eibit"
+#define D_TM_EXABYTE			"EB"
+#define D_TM_EXBIBYTE			"EiB"
+
+#define D_TM_ZETTABIT			"Zbit"
+#define D_TM_ZEBIBIT			"Zibit"
+#define D_TM_ZETTABYTE			"ZB"
+#define D_TM_ZEBIBYTE			"ZiB"
+
+#define D_TM_YOTTABIT			"Ybit"
+#define D_TM_YOBIBIT			"Yibit"
+#define D_TM_YOTTABYTE			"YB"
+#define D_TM_YOBIBYTE			"YiB"
+
+/** Units of measure for data rates */
+#define D_TM_BIT_PER_SECOND		D_TM_BIT"/s"
+#define D_TM_BYTE_PER_SECOND		D_TM_BYTE"/s"
+
+#define D_TM_KILOBIT_PER_SECOND		D_TM_KILOBIT"/s"
+#define D_TM_KIBIBIT_PER_SECOND		D_TM_KIBIBIT"/s"
+#define D_TM_KILOBYTE_PER_SECOND	D_TM_KILOBYTE"/s"
+#define D_TM_KIBIBYTE_PER_SECOND	D_TM_KIBIBYTE"/s"
+
+#define D_TM_MEGABIT_PER_SECOND		D_TM_MEGABIT"/s"
+#define D_TM_MEBIBIT_PER_SECOND		D_TM_MEBIBIT"/s"
+#define D_TM_MEGABYTE_PER_SECOND	D_TM_MEGABYTE"/s"
+#define D_TM_MEBIBYTE_PER_SECOND	D_TM_MEBIBYTE"/s"
+
+#define D_TM_GIGABIT_PER_SECOND		D_TM_GIGABIT"/s"
+#define D_TM_GIGIBIT_PER_SECOND		D_TM_GIGIBIT"/s"
+#define D_TM_GIGABYTE_PER_SECOND	D_TM_GIGABYTE"/s"
+#define D_TM_GIGIBYTE_PER_SECOND	D_TM_GIGIBYTE"/s"
+
+#define D_TM_TERABIT_PER_SECOND		D_TM_TERABIT"/s"
+#define D_TM_TEBIBIT_PER_SECOND		D_TM_TEBIBIT"/s"
+#define D_TM_TERABYTE_PER_SECOND	D_TM_TERABYTE"/s"
+#define D_TM_TEBIBYTE_PER_SECOND	D_TM_TEBIBYTE"/s"
+
+#define D_TM_PETABIT_PER_SECOND		D_TM_PETABIT"/s"
+#define D_TM_PEBIBIT_PER_SECOND		D_TM_PEBIBIT"/s"
+#define D_TM_PETABYTE_PER_SECOND	D_TM_PETABYTE"/s"
+#define D_TM_PEBIBYTE_PER_SECOND	D_TM_PEBIBYTE"/s"
+
+#define D_TM_EXABIT_PER_SECOND		D_TM_EXABIT"/s"
+#define D_TM_EXBIBIT_PER_SECOND		D_TM_EXBIBIT"s"
+#define D_TM_EXABYTE_PER_SECOND		D_TM_EXABYTE"/s"
+#define D_TM_EXBIBYTE_PER_SECOND	D_TM_EXBIBYTE"/s"
+
+#define D_TM_ZETTABIT_PER_SECOND	D_TM_ZETTABIT"/s"
+#define D_TM_ZEBIBIT_PER_SECOND		D_TM_ZEBIBIT"/s"
+#define D_TM_ZETTABYTE_PER_SECOND	D_TM_ZETTABYTE"/s"
+#define D_TM_ZEBIBYTE_PER_SECOND	D_TM_ZEBIBYTE"/s"
+
+#define D_TM_YOTTABIT_PER_SECOND	D_TM_YOTTABIT"/s"
+#define D_TM_YOBIBIT_PER_SECOND		D_TM_YOBIBIT"/s"
+#define D_TM_YOTTABYTE_PER_SECOND	D_TM_YOTTABYTE"/s"
+#define D_TM_YOBIBYTE_PER_SECOND	D_TM_YOBIBYTE"/s"
```

Comment:
Wow, that's really future-proofing. :)

Reply:
I was hoping it would last for a few years.  Besides, if I stopped sooner, someone would have said, "What about adding ... "